### ISSUE TITLE
The "git push" documentation (and maybe others) are broken due to include errors

### ISSUE THREAD

**Issue by avar** [git/git-scm.com#749]
See:

https://git-scm.com/docs/git-push#OPTIONS

If you click on the "GIT URLS" link it's broken. If you search for
"urls.txt" in the document you can see that this is because whatever
git-scm.org is using for its HTML generation apparently doesn't
understand recursive includes, git-push includes urls-remote.txt,
which in turn includes urls.txt, only the first level of includes is
processed.

The git.git "install-html" version of the documentation doesn't have this issue.

**Comment by peff** [comment]
git-scm uses asciidoctor to render the HTML. I checked and it _does_ handle nested includes just fine.

I think the problem is that the site itself preloads all of the necessary blobs into the database via a rake task ([here](https://github.com/git/git-scm.com/blob/57be1066b3e8ab0be0595e32222235817216685e/lib/tasks/local_index.rake#L65)), and that function checks for includes but does not recurse.

**Comment by peff** [comment]
IOW, something like this:

```diff
diff --git a/lib/tasks/local_index.rake b/lib/tasks/local_index.rake
index 254c3b4..3285c60 100644
--- a/lib/tasks/local_index.rake
+++ b/lib/tasks/local_index.rake
@@ -61,11 +61,14 @@ task :local_index => :environment do
 
         puts "   build: #{path}"
 
-        content = `git cat-file blob #{sha}`.chomp
-        content.gsub!(/include::(\S+)\.txt/) do |line|
-          line.gsub!("include::", "")
-          `git cat-file blob #{tag}:Documentation/#{line}`
+        def expand_include(tag, name)
+                content = `git cat-file blob #{name}`.chomp
+                content.gsub!(/include::(\S+)\.txt/) do |line|
+                        line.gsub!("include::", "")
+                        expand_include(tag, "#{tag}:Documentation/#{line}")
+                end
         end
+        content = expand_include(tag, sha)
         asciidoc = Asciidoctor::Document.new(content, templates_dir: template_dir)
         asciidoc_sha = Digest::SHA1.hexdigest( asciidoc.source )
```

but the end result does not seem quite right (I verified that it follows the recursion, but the result does not seem to format it correctly. It's probably something simple and stupid, but I'm not seeing it).

**Comment by peff** [comment]
Closed by #750.
